CMakeLists.txt

```diff
@@ -784,16 +784,27 @@ endif()
 
 if(AVIF_BUILD_APPS)
     add_executable(avifenc apps/avifenc.c)
-    if(AVIF_USE_CXX)
-        set_target_properties(avifenc PROPERTIES LINKER_LANGUAGE "CXX")
-    endif()
-    target_link_libraries(avifenc avif_apps)
     add_executable(avifdec apps/avifdec.c)
-    if(AVIF_USE_CXX)
-        set_target_properties(avifdec PROPERTIES LINKER_LANGUAGE "CXX")
-    endif()
-    target_link_libraries(avifdec avif_apps)
-
+    foreach(PROG IN ITEMS avifenc avifdec)
+        if(AVIF_USE_CXX)
+            set_target_properties(${PROG} PROPERTIES LINKER_LANGUAGE "CXX")
+        endif()
+        target_link_libraries(${PROG} avif_apps)
+        if(WIN32)
+            # Add a manifest file to deal with UTF8.
+            if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+                target_sources(${PROG} PUBLIC ${CMAKE_SOURCE_DIR}/src/utf8.manifest)
+            else()
+                add_custom_command(
+                    TARGET ${PROG}
+                    POST_BUILD
+                    COMMAND "mt.exe" -manifest \"${CMAKE_SOURCE_DIR}\\src\\utf8.manifest\"
```

Comment:
I don't think this can work on MinGW, there is no mt.exe...

Reply:
Can MINGW access the Windows SDK and therefore "mt.exe"?
It seems adding the manifest as source (or at least the .rc) works, cf https://doc.magnum.graphics/magnum/platforms-windows.html . Can you confirm this is the way to go? I have no experience with manifest files, @wantehchang just found this cleaner solution for handling UTF8.

Reply:
I don't think one can assume MinGW users/devs will have access to the Windows SDK.

I might be completely wrong here, but perhaps a conversation is to be had w/ the MinGW (or is it Cygwin?) project to ship a pre-built _UTF8_ object manifest that can be statically linked in then, [like the existing ("regular"?) one](https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-windows-default-manifest?repo=ucrt64)?

Reply:
Furthermore, the https://cygwin.com/git/?p=cygwin-apps/windows-default-manifest.git;a=tree also seems to wrap that default manifest into a .rc file, and resource files can be dealt with w/o a problem in MinGW toolchains (windres is shipped). Maybe that could be a temporary workaround?

Reply:
Maybe take a [lead ](https://blog.r-project.org/2022/11/07/issues-while-switching-r-to-utf-8-and-ucrt-on-windows/index.html)from [RStudio](https://github.com/rstudio/rstudio/blob/a1d6917afc68965cfb7a964a9afcfcb04da0e7d7/src/cpp/session/CMakeLists.txt#L466)?

Reply:
Miloš: Thank you very much for the research.

The UTF-8 support for libavif is simpler than the UTF-8 support for RStudio, because the libavif library itself has only one function, `avifIOCreateFileReader()`, that takes a `filename` input parameter, and that function faithfully passes `filename` to `fopen()` with no processing. So we can document that the libavif library honors the code pages selected by the application, and we just need to solve the UTF-8 support problem for the three command-line programs, avifdec, avifenc, and are_images_equal. And it is fine to solve the UTF-8 support problem on Windows 10 and later only so that we can use the simple solution of the manifest file.